### PR TITLE
Add support for DB URL connection strings

### PR DESCRIPTION
**Pull request by mbryzek**
Modifies sem-apply to accept a --url parameter containing a connection string
for the psql database to which we are applying changes.

Connection string format follows psql:

 - Example: postgresql://postgres@localhost/sample_development
 - See https://jdbc.postgresql.org/documentation/head/connect.html

Original proposal at https://github.com/mbryzek/schema-evolution-manager/pull/21

### CODE REVIEW

**Comment by Seanstoppable** [comment]
+1

Particularly like the refactor of the internals to just use url, no matter the params.

File: bin/sem-init [hunk 1]
```diff
@@ -1,24 +1,23 @@
...
+#  bin/sem-init --dir /tmp/example_repo --url postgresql://username@localhost/sample_development
 #
 
 load File.join(File.dirname(__FILE__), 'sem-config')
 SchemaEvolutionManager::Library.set_verbose(true)
 
-args = SchemaEvolutionManager::Args.from_stdin(:required => %w(name user dir))
-SchemaEvolutionManager::Preconditions.check_state(File.directory?(args.dir), "Dir[%s] does not exist" % [args.dir])
+args = SchemaEvolutionManager::Args.from_stdin(:required => %w(dir url))
+SchemaEvolutionManager::Preconditions.check_state(File.directory?(args.dir), "Dir[%s] does not exist" % args.dir)
```
Comment (mfine) [inline bin/sem-init]: Ok with me, but it might come off as inconsistent to not support `user`, `host` here vs. `sem-apply`.

**Comment by mfine** [comment]
Looks great, thanks!

**Comment by mbryzek** [comment]
Have updated sem-init to support both parameters. Merging now

**Comment by mbryzek** [comment]
Released as 0.9.20

**Pull request merged**